Move unversioned.GroupVersion to pkg/runtime/schema

smarterclayton opened the pull request:
@caesarxuchao extracted from #36121, will my refactor PR after yours

@thockin as discussed about breaking circular dependencies.  More will follow

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36782)
<!-- Reviewable:end -->

smarterclayton committed:
Move GroupVersion* to pkg/runtime/schema

caesarxuchao commented on convert.sh:
```diff
@@ -0,0 +1,36 @@
+echo pkg/client/typed/discovery/helper_blackbox_test.go | xargs -n1 sed -i "s|uapi\\.GroupVersion|schema\\.GroupVersion|g"
+cat ${files_to_convert} | xargs -n1 sed -i "s|schema\\.GroupVersionForDiscovery|unversioned\\.GroupVersionForDiscovery|g"
+goimports -w ./pkg ./federation ./test ./plugin ./cmd
+echo test/e2e_node/apparmor_test.go | xargs -n1 sed -i "s|k8s\\.io/kubernetes/pkg/runtime/schema|k8s\\.io/client-go/pkg/runtime/schema|g"
```
Why client-go?

smarterclayton replied:
Errors package in that test comes from client-go, you must pass runtime/schema from client package to that method.

caesarxuchao replied:
I confirmed with @timstclair client-go was imported by accident. Suggest import the kubernetes packages.

thockin commented:
I tagged you on the PR for a simple way for me to break the cycle, too.
Much smaller than this PR :)

2016-11-14 15:25 GMT-08:00 Clayton Coleman <notifications@github.com>:

> @caesarxuchao <https://github.com/caesarxuchao> extracted from #36121
> <https://github.com/kubernetes/kubernetes/pull/36121>, will my refactor
> PR after yours
>
> @thockin <https://github.com/thockin> as discussed about breaking
> circular dependencies. More will follow
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/kubernetes/kubernetes/pull/36782
> Commit Summary
>
>    - Move GroupVersion* to pkg/runtime/schema
>    - generated: refactored code
>
> File Changes
>
>    - *M* cmd/kube-apiserver/app/server.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-0> (6)
>    - *M* cmd/kube-controller-manager/app/controllermanager.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-1> (5)
>    - *M* cmd/kube-dns/app/server.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-2> (4)
>    - *M* cmd/kubeadm/app/apis/kubeadm/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-3>
>    (14)
>    - *M* cmd/kubeadm/app/apis/kubeadm/v1alpha1/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-4>
>    (14)
>    - *M* cmd/libs/go2idl/client-gen/test_apis/testgroup/install/install.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-5>
>    (12)
>    - *M* cmd/libs/go2idl/client-gen/test_apis/testgroup/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-6> (8)
>    - *M* cmd/libs/go2idl/client-gen/test_apis/testgroup/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-7> (7)
>    - *M* cmd/libs/go2idl/client-gen/test_apis/testgroup/v1/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-8> (7)
>    - *M* cmd/libs/go2idl/client-gen/test_apis/testgroup/v1/types.
>    generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-9> (7)
>    - *M* cmd/libs/go2idl/client-gen/testoutput/clientset_generated/test_
>    internalclientset/typed/testgroup/internalversion/fake/fake_testtype.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-10>
>    (4)
>    - *M* cmd/libs/go2idl/go-to-protobuf/protobuf/cmd.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-11>
>    (1)
>    - *A* convert.sh
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-12>
>    (19)
>    - *M* federation/apis/core/install/install.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-13>
>    (14)
>    - *M* federation/apis/core/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-14>
>    (9)
>    - *M* federation/apis/core/v1/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-15>
>    (3)
>    - *M* federation/apis/federation/install/install.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-16>
>    (14)
>    - *M* federation/apis/federation/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-17>
>    (12)
>    - *M* federation/apis/federation/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-18>
>    (7)
>    - *M* federation/apis/federation/v1beta1/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-19>
>    (8)
>    - *M* federation/apis/federation/v1beta1/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-20>
>    (7)
>    - *M* federation/apis/federation/v1beta1/zz_generated.conversion.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-21>
>    (3)
>    - *M* federation/apis/federation/v1beta1/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-22>
>    (3)
>    - *M* federation/apis/federation/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-23>
>    (3)
>    - *M* federation/client/clientset_generated/federation_
>    internalclientset/typed/core/internalversion/fake/fake_configmap.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-24>
>    (4)
>    - *M* federation/client/clientset_generated/federation_
>    internalclientset/typed/core/internalversion/fake/fake_event.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-25>
>    (4)
>    - *M* federation/client/clientset_generated/federation_
>    internalclientset/typed/core/internalversion/fake/fake_namespace.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-26>
>    (4)
>    - *M* federation/client/clientset_generated/federation_
>    internalclientset/typed/core/internalversion/fake/fake_secret.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-27>
>    (4)
>    - *M* federation/client/clientset_generated/federation_
>    internalclientset/typed/core/internalversion/fake/fake_service.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-28>
>    (4)
>    - *M* federation/client/clientset_generated/federation_
>    internalclientset/typed/extensions/internalversion/
>    fake/fake_daemonset.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-29>
>    (4)
>    - *M* federation/client/clientset_generated/federation_
>    internalclientset/typed/extensions/internalversion/
>    fake/fake_deployment.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-30>
>    (4)
>    - *M* federation/client/clientset_generated/federation_
>    internalclientset/typed/extensions/internalversion/fake/fake_ingress.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-31>
>    (4)
>    - *M* federation/client/clientset_generated/federation_
>    internalclientset/typed/extensions/internalversion/
>    fake/fake_replicaset.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-32>
>    (4)
>    - *M* federation/client/clientset_generated/federation_
>    internalclientset/typed/federation/internalversion/fake/fake_cluster.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-33>
>    (4)
>    - *M* federation/client/clientset_generated/federation_release_
>    1_5/typed/core/v1/core_client.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-34>
>    (5)
>    - *M* federation/client/clientset_generated/federation_release_
>    1_5/typed/core/v1/fake/fake_configmap.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-35>
>    (4)
>    - *M* federation/client/clientset_generated/federation_release_
>    1_5/typed/core/v1/fake/fake_event.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-36>
>    (4)
>    - *M* federation/client/clientset_generated/federation_release_
>    1_5/typed/core/v1/fake/fake_namespace.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-37>
>    (4)
>    - *M* federation/client/clientset_generated/federation_release_
>    1_5/typed/core/v1/fake/fake_secret.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-38>
>    (4)
>    - *M* federation/client/clientset_generated/federation_release_
>    1_5/typed/core/v1/fake/fake_service.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-39>
>    (4)
>    - *M* federation/client/clientset_generated/federation_release_
>    1_5/typed/extensions/v1beta1/extensions_client.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-40>
>    (5)
>    - *M* federation/client/clientset_generated/federation_release_
>    1_5/typed/extensions/v1beta1/fake/fake_daemonset.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-41>
>    (4)
>    - *M* federation/client/clientset_generated/federation_release_
>    1_5/typed/extensions/v1beta1/fake/fake_deployment.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-42>
>    (4)
>    - *M* federation/client/clientset_generated/federation_release_
>    1_5/typed/extensions/v1beta1/fake/fake_ingress.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-43>
>    (4)
>    - *M* federation/client/clientset_generated/federation_release_
>    1_5/typed/extensions/v1beta1/fake/fake_replicaset.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-44>
>    (4)
>    - *M* federation/client/clientset_generated/federation_release_
>    1_5/typed/federation/v1beta1/fake/fake_cluster.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-45>
>    (4)
>    - *M* federation/client/clientset_generated/federation_release_
>    1_5/typed/federation/v1beta1/federation_client.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-46>
>    (5)
>    - *M* federation/cmd/federation-apiserver/app/server.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-47>
>    (8)
>    - *M* federation/pkg/dnsprovider/providers/aws/route53/zone.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-48>
>    (3)
>    - *M* federation/pkg/dnsprovider/providers/google/clouddns/zone.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-49>
>    (3)
>    - *M* federation/pkg/federation-controller/service/dns.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-50>
>    (3)
>    - *M* federation/pkg/federation-controller/service/dns_test.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-51>
>    (5)
>    - *M* federation/pkg/kubefed/unjoin.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-52>
>    (4)
>    - *M* pkg/admission/attributes.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-53>
>    (12)
>    - *M* pkg/admission/chain_test.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-54>
>    (4)
>    - *M* pkg/admission/errors.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-55>
>    (6)
>    - *M* pkg/admission/interfaces.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-56>
>    (6)
>    - *M* pkg/api/copy_test.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-57>
>    (8)
>    - *M* pkg/api/defaulting_test.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-58>
>    (7)
>    - *M* pkg/api/errors/errors.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-59>
>    (21)
>    - *M* pkg/api/errors/errors_test.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-60>
>    (3)
>    - *M* pkg/api/errors/storage/storage.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-61>
>    (16)
>    - *M* pkg/api/install/install.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-62>
>    (14)
>    - *M* pkg/api/install/install_test.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-63>
>    (3)
>    - *M* pkg/api/mapper.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-64>
>    (6)
>    - *M* pkg/api/meta/errors.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-65>
>    (18)
>    - *M* pkg/api/meta/firsthit_restmapper.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-66>
>    (12)
>    - *M* pkg/api/meta/help_test.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-67>
>    (10)
>    - *M* pkg/api/meta/interfaces.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-68>
>    (15)
>    - *M* pkg/api/meta/meta.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-69>
>    (5)
>    - *M* pkg/api/meta/meta_test.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-70>
>    (13)
>    - *M* pkg/api/meta/multirestmapper.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-71>
>    (28)
>    - *M* pkg/api/meta/multirestmapper_test.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-72>
>    (156)
>    - *M* pkg/api/meta/priority.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-73>
>    (44)
>    - *M* pkg/api/meta/priority_test.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-74>
>    (110)
>    - *M* pkg/api/meta/restmapper.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-75>
>    (78)
>    - *M* pkg/api/meta/restmapper_test.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-76>
>    (216)
>    - *M* pkg/api/meta/unstructured.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-77>
>    (4)
>    - *M* pkg/api/meta_test.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-78>
>    (5)
>    - *M* pkg/api/ref.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-79>
>    (10)
>    - *M* pkg/api/ref_test.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-80>
>    (5)
>    - *M* pkg/api/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-81>
>    (9)
>    - *M* pkg/api/rest/create.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-82>
>    (8)
>    - *M* pkg/api/rest/delete.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-83>
>    (3)
>    - *M* pkg/api/rest/rest.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-84>
>    (3)
>    - *M* pkg/api/serialization_proto_test.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-85>
>    (6)
>    - *M* pkg/api/serialization_test.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-86>
>    (11)
>    - *M* pkg/api/testapi/testapi.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-87>
>    (54)
>    - *M* pkg/api/testing/compat/compatibility_tester.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-88>
>    (4)
>    - *M* pkg/api/testing/fuzzer.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-89>
>    (5)
>    - *M* pkg/api/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-90>
>    (7)
>    - *M* pkg/api/unversioned/generated.pb.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-91>
>    (10)
>    - *M* pkg/api/unversioned/group_version.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-92>
>    (205)
>    - *M* pkg/api/unversioned/meta.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-93>
>    (12)
>    - *M* pkg/api/unversioned/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-94>
>    (8)
>    - *M* pkg/api/unversioned/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-95>
>    (3)
>    - *M* pkg/api/v1/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-96>
>    (3)
>    - *M* pkg/api/v1/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-97>
>    (7)
>    - *M* pkg/api/v1/zz_generated.conversion.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-98>
>    (3)
>    - *M* pkg/api/v1/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-99>
>    (3)
>    - *M* pkg/api/validation/events.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-100>
>    (4)
>    - *M* pkg/api/validation/validation.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-101>
>    (6)
>    - *M* pkg/api/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-102>
>    (3)
>    - *M* pkg/apimachinery/announced/group_factory.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-103>
>    (18)
>    - *M* pkg/apimachinery/registered/registered.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-104>
>    (80)
>    - *M* pkg/apimachinery/registered/registered_test.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-105>
>    (16)
>    - *M* pkg/apimachinery/types.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-106>
>    (16)
>    - *M* pkg/apimachinery/types_test.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-107>
>    (8)
>    - *M* pkg/apis/abac/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-108>
>    (6)
>    - *M* pkg/apis/abac/v0/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-109>
>    (6)
>    - *M* pkg/apis/abac/v1beta1/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-110>
>    (6)
>    - *M* pkg/apis/apps/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-111>
>    (8)
>    - *M* pkg/apis/apps/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-112>
>    (7)
>    - *M* pkg/apis/apps/v1beta1/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-113>
>    (8)
>    - *M* pkg/apis/apps/v1beta1/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-114>
>    (7)
>    - *M* pkg/apis/apps/v1beta1/zz_generated.conversion.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-115>
>    (3)
>    - *M* pkg/apis/apps/v1beta1/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-116>
>    (3)
>    - *M* pkg/apis/apps/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-117>
>    (3)
>    - *M* pkg/apis/authentication/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-118>
>    (8)
>    - *M* pkg/apis/authentication/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-119>
>    (7)
>    - *M* pkg/apis/authentication/v1beta1/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-120>
>    (4)
>    - *M* pkg/apis/authentication/v1beta1/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-121>
>    (7)
>    - *M* pkg/apis/authentication/v1beta1/zz_generated.conversion.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-122>
>    (3)
>    - *M* pkg/apis/authentication/v1beta1/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-123>
>    (3)
>    - *M* pkg/apis/authentication/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-124>
>    (3)
>    - *M* pkg/apis/authorization/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-125>
>    (8)
>    - *M* pkg/apis/authorization/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-126>
>    (7)
>    - *M* pkg/apis/authorization/v1beta1/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-127>
>    (10)
>    - *M* pkg/apis/authorization/v1beta1/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-128>
>    (7)
>    - *M* pkg/apis/authorization/v1beta1/zz_generated.conversion.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-129>
>    (3)
>    - *M* pkg/apis/authorization/v1beta1/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-130>
>    (3)
>    - *M* pkg/apis/authorization/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-131>
>    (3)
>    - *M* pkg/apis/autoscaling/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-132>
>    (8)
>    - *M* pkg/apis/autoscaling/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-133>
>    (7)
>    - *M* pkg/apis/autoscaling/v1/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-134>
>    (4)
>    - *M* pkg/apis/autoscaling/v1/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-135>
>    (7)
>    - *M* pkg/apis/autoscaling/v1/zz_generated.conversion.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-136>
>    (3)
>    - *M* pkg/apis/autoscaling/v1/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-137>
>    (3)
>    - *M* pkg/apis/autoscaling/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-138>
>    (3)
>    - *M* pkg/apis/batch/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-139>
>    (8)
>    - *M* pkg/apis/batch/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-140>
>    (7)
>    - *M* pkg/apis/batch/v1/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-141>
>    (4)
>    - *M* pkg/apis/batch/v1/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-142>
>    (7)
>    - *M* pkg/apis/batch/v1/zz_generated.conversion.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-143>
>    (3)
>    - *M* pkg/apis/batch/v1/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-144>
>    (3)
>    - *M* pkg/apis/batch/v2alpha1/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-145>
>    (4)
>    - *M* pkg/apis/batch/v2alpha1/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-146>
>    (7)
>    - *M* pkg/apis/batch/v2alpha1/zz_generated.conversion.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-147>
>    (3)
>    - *M* pkg/apis/batch/v2alpha1/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-148>
>    (3)
>    - *M* pkg/apis/batch/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-149>
>    (3)
>    - *M* pkg/apis/certificates/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-150>
>    (12)
>    - *M* pkg/apis/certificates/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-151>
>    (7)
>    - *M* pkg/apis/certificates/v1alpha1/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-152>
>    (12)
>    - *M* pkg/apis/certificates/v1alpha1/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-153>
>    (7)
>    - *M* pkg/apis/certificates/v1alpha1/zz_generated.conversion.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-154>
>    (3)
>    - *M* pkg/apis/certificates/v1alpha1/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-155>
>    (3)
>    - *M* pkg/apis/certificates/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-156>
>    (3)
>    - *M* pkg/apis/componentconfig/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-157>
>    (14)
>    - *M* pkg/apis/componentconfig/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-158>
>    (7)
>    - *M* pkg/apis/componentconfig/v1alpha1/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-159>
>    (10)
>    - *M* pkg/apis/componentconfig/v1alpha1/zz_generated.conversion.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-160>
>    (3)
>    - *M* pkg/apis/componentconfig/v1alpha1/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-161>
>    (3)
>    - *M* pkg/apis/componentconfig/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-162>
>    (3)
>    - *M* pkg/apis/extensions/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-163>
>    (8)
>    - *M* pkg/apis/extensions/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-164>
>    (7)
>    - *M* pkg/apis/extensions/v1beta1/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-165>
>    (4)
>    - *M* pkg/apis/extensions/v1beta1/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-166>
>    (7)
>    - *M* pkg/apis/extensions/v1beta1/zz_generated.conversion.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-167>
>    (3)
>    - *M* pkg/apis/extensions/v1beta1/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-168>
>    (3)
>    - *M* pkg/apis/extensions/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-169>
>    (3)
>    - *M* pkg/apis/imagepolicy/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-170>
>    (8)
>    - *M* pkg/apis/imagepolicy/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-171>
>    (7)
>    - *M* pkg/apis/imagepolicy/v1alpha1/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-172>
>    (4)
>    - *M* pkg/apis/imagepolicy/v1alpha1/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-173>
>    (7)
>    - *M* pkg/apis/imagepolicy/v1alpha1/zz_generated.conversion.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-174>
>    (3)
>    - *M* pkg/apis/imagepolicy/v1alpha1/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-175>
>    (3)
>    - *M* pkg/apis/imagepolicy/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-176>
>    (3)
>    - *M* pkg/apis/policy/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-177>
>    (8)
>    - *M* pkg/apis/policy/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-178>
>    (7)
>    - *M* pkg/apis/policy/v1alpha1/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-179>
>    (4)
>    - *M* pkg/apis/policy/v1beta1/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-180>
>    (4)
>    - *M* pkg/apis/policy/v1beta1/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-181>
>    (7)
>    - *M* pkg/apis/policy/v1beta1/zz_generated.conversion.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-182>
>    (3)
>    - *M* pkg/apis/policy/v1beta1/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-183>
>    (3)
>    - *M* pkg/apis/policy/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-184>
>    (3)
>    - *M* pkg/apis/rbac/helpers.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-185>
>    (6)
>    - *M* pkg/apis/rbac/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-186>
>    (8)
>    - *M* pkg/apis/rbac/v1alpha1/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-187>
>    (4)
>    - *M* pkg/apis/rbac/v1alpha1/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-188>
>    (7)
>    - *M* pkg/apis/rbac/v1alpha1/zz_generated.conversion.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-189>
>    (3)
>    - *M* pkg/apis/rbac/v1alpha1/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-190>
>    (3)
>    - *M* pkg/apis/rbac/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-191>
>    (3)
>    - *M* pkg/apis/storage/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-192>
>    (8)
>    - *M* pkg/apis/storage/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-193>
>    (7)
>    - *M* pkg/apis/storage/v1beta1/register.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-194>
>    (4)
>    - *M* pkg/apis/storage/v1beta1/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-195>
>    (7)
>    - *M* pkg/apis/storage/v1beta1/zz_generated.conversion.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-196>
>    (3)
>    - *M* pkg/apis/storage/v1beta1/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-197>
>    (3)
>    - *M* pkg/apis/storage/zz_generated.deepcopy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-198>
>    (3)
>    - *M* pkg/apiserver/api_installer.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-199>
>    (13)
>    - *M* pkg/apiserver/apiserver.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-200>
>    (27)
>    - *M* pkg/apiserver/apiserver_test.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-201>
>    (43)
>    - *M* pkg/apiserver/errors_test.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-202>
>    (3)
>    - *M* pkg/apiserver/negotiate.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-203>
>    (18)
>    - *M* pkg/apiserver/proxy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-204>
>    (6)
>    - *M* pkg/apiserver/resthandler.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-205>
>    (15)
>    - *M* pkg/apiserver/resthandler_test.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-206>
>    (5)
>    - *M* pkg/apiserver/serviceerror.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-207>
>    (4)
>    - *M* pkg/apiserver/testing/types.generated.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-208>
>    (7)
>    - *M* pkg/apiserver/testing/types.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-209>
>    (9)
>    - *M* pkg/client/cache/listers.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-210>
>    (7)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/apps/internalversion/fake/fake_statefulset.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-211>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/autoscaling/internalversion/fake/fake_horizontalpodautoscaler.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-212>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/batch/internalversion/fake/fake_cronjob.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-213>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/batch/internalversion/fake/fake_job.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-214>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/certificates/internalversion/fake/fake_
>    certificatesigningrequest.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-215>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/core/internalversion/fake/fake_componentstatus.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-216>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/core/internalversion/fake/fake_configmap.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-217>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/core/internalversion/fake/fake_endpoints.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-218>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/core/internalversion/fake/fake_event.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-219>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/core/internalversion/fake/fake_limitrange.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-220>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/core/internalversion/fake/fake_namespace.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-221>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/core/internalversion/fake/fake_node.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-222>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/core/internalversion/fake/fake_persistentvolume.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-223>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/core/internalversion/fake/fake_persistentvolumeclaim.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-224>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/core/internalversion/fake/fake_pod.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-225>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/core/internalversion/fake/fake_podtemplate.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-226>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/core/internalversion/fake/fake_replicationcontroller.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-227>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/core/internalversion/fake/fake_resourcequota.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-228>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/core/internalversion/fake/fake_secret.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-229>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/core/internalversion/fake/fake_service.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-230>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/core/internalversion/fake/fake_serviceaccount.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-231>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/extensions/internalversion/fake/fake_daemonset.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-232>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/extensions/internalversion/fake/fake_deployment.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-233>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/extensions/internalversion/fake/fake_ingress.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-234>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/extensions/internalversion/fake/fake_networkpolicy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-235>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/extensions/internalversion/fake/fake_podsecuritypolicy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-236>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/extensions/internalversion/fake/fake_replicaset.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-237>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/extensions/internalversion/fake/fake_scale_expansion.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-238>
>    (6)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/extensions/internalversion/fake/fake_thirdpartyresource.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-239>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/extensions/internalversion/scale_expansion.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-240>
>    (6)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/policy/internalversion/fake/fake_eviction_expansion.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-241>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/policy/internalversion/fake/fake_poddisruptionbudget.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-242>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/rbac/internalversion/fake/fake_clusterrole.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-243>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/rbac/internalversion/fake/fake_clusterrolebinding.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-244>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/rbac/internalversion/fake/fake_role.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-245>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/rbac/internalversion/fake/fake_rolebinding.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-246>
>    (4)
>    - *M* pkg/client/clientset_generated/internalclientset/
>    typed/storage/internalversion/fake/fake_storageclass.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-247>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    apps/v1beta1/apps_client.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-248>
>    (5)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    apps/v1beta1/fake/fake_statefulset.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-249>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    authentication/v1beta1/authentication_client.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-250>
>    (5)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    authorization/v1beta1/authorization_client.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-251>
>    (5)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    autoscaling/v1/autoscaling_client.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-252>
>    (5)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    autoscaling/v1/fake/fake_horizontalpodautoscaler.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-253>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    batch/v1/batch_client.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-254>
>    (5)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    batch/v1/fake/fake_job.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-255>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    batch/v2alpha1/batch_client.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-256>
>    (5)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    batch/v2alpha1/fake/fake_cronjob.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-257>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    batch/v2alpha1/fake/fake_job.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-258>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    certificates/v1alpha1/certificates_client.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-259>
>    (5)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    certificates/v1alpha1/fake/fake_certificatesigningrequest.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-260>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    core/v1/core_client.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-261>
>    (5)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    core/v1/fake/fake_componentstatus.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-262>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    core/v1/fake/fake_configmap.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-263>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    core/v1/fake/fake_endpoints.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-264>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    core/v1/fake/fake_event.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-265>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    core/v1/fake/fake_limitrange.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-266>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    core/v1/fake/fake_namespace.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-267>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    core/v1/fake/fake_node.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-268>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    core/v1/fake/fake_persistentvolume.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-269>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    core/v1/fake/fake_persistentvolumeclaim.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-270>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    core/v1/fake/fake_pod.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-271>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    core/v1/fake/fake_podtemplate.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-272>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    core/v1/fake/fake_replicationcontroller.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-273>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    core/v1/fake/fake_resourcequota.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-274>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    core/v1/fake/fake_secret.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-275>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    core/v1/fake/fake_service.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-276>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    core/v1/fake/fake_serviceaccount.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-277>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    extensions/v1beta1/extensions_client.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-278>
>    (5)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    extensions/v1beta1/fake/fake_daemonset.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-279>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    extensions/v1beta1/fake/fake_deployment.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-280>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    extensions/v1beta1/fake/fake_ingress.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-281>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    extensions/v1beta1/fake/fake_job.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-282>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    extensions/v1beta1/fake/fake_podsecuritypolicy.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-283>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    extensions/v1beta1/fake/fake_replicaset.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-284>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    extensions/v1beta1/fake/fake_scale_expansion.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-285>
>    (6)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    extensions/v1beta1/fake/fake_thirdpartyresource.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-286>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    extensions/v1beta1/scale_expansion.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-287>
>    (6)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    policy/v1alpha1/fake/fake_poddisruptionbudget.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-288>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    policy/v1alpha1/policy_client.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-289>
>    (5)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    policy/v1beta1/fake/fake_poddisruptionbudget.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-290>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    policy/v1beta1/policy_client.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-291>
>    (5)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    rbac/v1alpha1/fake/fake_clusterrole.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-292>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    rbac/v1alpha1/fake/fake_clusterrolebinding.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-293>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    rbac/v1alpha1/fake/fake_role.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-294>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    rbac/v1alpha1/fake/fake_rolebinding.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-295>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    rbac/v1alpha1/rbac_client.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-296>
>    (5)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    storage/v1beta1/fake/fake_storageclass.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-297>
>    (4)
>    - *M* pkg/client/clientset_generated/release_1_5/typed/
>    storage/v1beta1/storage_client.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-298>
>    (5)
>    - *M* pkg/client/restclient/client.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-299>
>    (10)
>    - *I* pkg/client/restclient/client_test.go
>    <https://github.com/kubernetes/kubernetes/pull/36782/files#diff-300>
>    (0)
>
> Patch Links:
>
>    - https://github.com/kubernetes/kubernetes/pull/36782.patch
>    - https://github.com/kubernetes/kubernetes/pull/36782.diff
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/36782>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFVgVGjRlnzamE1ekTHoDFHgYNYPIXHTks5q-O32gaJpZM4Kx79x>
> .
>

smarterclayton committed:
generated: refactor

smarterclayton committed:
generated: client to staging

caesarxuchao commented:
@smarterclayton the changes look good. How much more do you have locally?

smarterclayton commented:
I rebased on top of your branch and dropped convert.sh out as well as fixing a few small things that the internal -> versioned change caused.  I'll also drop the "staging/copy.sh on Mac" commit since it's possible that breaks Linux and it was just local.

caesarxuchao commented:
e2e:
```
W1123 13:31:47.898] test/e2e/network_partition.go:371: cannot use StatefulSetGroupVersionResource (type "k8s.io/kubernetes/pkg/api/unversioned".GroupVersionResource) as type "k8s.io/kubernetes/pkg/runtime/schema".GroupVersionResource in argument to framework.SkipIfMissingResource
W1123 13:31:47.899] test/e2e/petset.go:83: cannot use StatefulSetGroupVersionResource (type "k8s.io/kubernetes/pkg/api/unversioned".GroupVersionResource) as type "k8s.io/kubernetes/pkg/runtime/schema".GroupVersionResource in argument to framework.SkipIfMissingResource
W1123 13:31:47.899] test/e2e/petset.go:411: cannot use StatefulSetGroupVersionResource (type "k8s.io/kubernetes/pkg/api/unversioned".GroupVersionResource) as type "k8s.io/kubernetes/pkg/runtime/schema".GroupVersionResource in argument to framework.SkipIfMissingResource
```

smarterclayton commented:
Yeah, I can't compile on mac right now because of the cross compile issue
so I suspect my script didn't catch it.

On Nov 23, 2016, at 3:38 PM, Chao Xu <notifications@github.com> wrote:

e2e:

W1123 13:31:47.898] test/e2e/network_partition.go:371: cannot use
StatefulSetGroupVersionResource (type
"k8s.io/kubernetes/pkg/api/unversioned".GroupVersionResource) as type
"k8s.io/kubernetes/pkg/runtime/schema".GroupVersionResource in
argument to framework.SkipIfMissingResource
W1123 13:31:47.899] test/e2e/petset.go:83: cannot use
StatefulSetGroupVersionResource (type
"k8s.io/kubernetes/pkg/api/unversioned".GroupVersionResource) as type
"k8s.io/kubernetes/pkg/runtime/schema".GroupVersionResource in
argument to framework.SkipIfMissingResource
W1123 13:31:47.899] test/e2e/petset.go:411: cannot use
StatefulSetGroupVersionResource (type
"k8s.io/kubernetes/pkg/api/unversioned".GroupVersionResource) as type
"k8s.io/kubernetes/pkg/runtime/schema".GroupVersionResource in
argument to framework.SkipIfMissingResource

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/36782#issuecomment-262636230>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/ABG_p8pL6q91wZDO1msJsL455Gqcsa_Zks5rBLJhgaJpZM4Kx79x>
.

smarterclayton commented:
Actually this is because your branch isn't up to date with latest.

saad-ali commented:
Ok I just merged Chao's PR https://github.com/kubernetes/kubernetes/pull/36673. Clayton, feel free to rebase. Ping me when you're done. I will be back online around 8:30 PM PST and will merge your PR then, if the queue is still green.

saad-ali commented:
@k8s-bot verify test this

smarterclayton commented:
There were actual verification failures, pushing another change now.

smarterclayton commented:
All green

saad-ali commented:
Merging
